pygam/pygam.py

```diff
@@ -780,7 +780,7 @@ def _pirls(self, X, Y, weights):
             self._on_loop_start(vars())
 
             WB = W.dot(modelmat[mask, :])  # common matrix product
-            Q, R = np.linalg.qr(WB.A)
+            Q, R = np.linalg.qr(WB.toarray())
```

Comment:
why do we change to array here?

Reply:
Scipy 1.14.0 deprecated the methods A. To keep functionality I change it to .toarray()

Reply:
I see. And why are we changing to scipy 1.14

Reply:
It's good practice to have your dependencies updated. Although when installed with pip it installs the latest version of scipy wich leads to the bug outline in #357